src/gauge/radial-gauge/radial-gauge.component.ts

```diff
@@ -0,0 +1,266 @@
+  @Input() public segments = [];
+
+  @Output() public click: EventEmitter<any> = new EventEmitter();
+  @Output() public mouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public mouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public arcClick: EventEmitter<any> = new EventEmitter();
+  @Output() public arcMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public arcMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public labelClick: EventEmitter<any> = new EventEmitter();
+  @Output() public labelMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public labelMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public pointerClick: EventEmitter<any> = new EventEmitter();
+  @Output() public pointerMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public pointerMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  private colors: ColorHelper;
+  private ticks: number[];
+  private degreeRange: number;
+
+  ngAfterViewInit(): void {
+    super.ngAfterViewInit();
+  }
+
+  public update(): void {
+    super.update();
+
+    this.ticks = this.getTicks();
+    this.degreeRange = this.getDegreeRange();
+
+    this.dimensions = this.getDimensions();
+
+    this.displayingValue = !this.checkStringEmpty(this.displayValue) 
+    ? this.displayValue 
+    : this.value.toString();
+
+    this.colors = new ColorHelper(this.scheme, this.schemeType, 
+      [this.minValue, this.maxValue], this.customColors);
+
+    this.arcs = this.getArcs();
+  }
+
+  public getTranslate(): string {
+    const twidth = this.width / 2;
+    const theight = this.height / 2;
+    return `translate(${twidth}, ${theight})`;
+  }
+
+  public movePointer(): string {
+    return `rotate(${this.getPointerLocation()})`;
+  }
+
+  public getTextTransform(): string {
+    const scale: number = this.getElementScale(0.004);
+    return `scale(${scale}, ${scale})`;
+  }
+
+  public getPointer(): string {
+    const pointerLine: Line<[number, number]> = line().curve(curveLinear);
+    return pointerLine(this.getPointerData());
+  }
+
+  public getScale(): ScaleLinear<number, number> {
+    return scaleLinear()
+    .range([0, this.getDegreeRange()])
+    .nice()
+    .domain([this.minValue, this.maxValue]);
+  }
+
+  public getArcs(): any {
+    const result = [];
+
+    if (this.segments != null && this.segments.length > 0) {
+      for(const i in this.segments) {
+        const arc = {
+          startAngle: this.getSegmentAngle(this.segments[i].minValue),
+          endAngle: this.getSegmentAngle(this.segments[i].maxValue),
+          innerRadius: this.dimensions.innerArcRadius,
+          outerRadius: this.dimensions.outerArcRadius,
+          color: this.getValueOr(this.segments[i].color, 
+            this.colors.getColor(this.segments[i].minValue))
+        };
+        result.push(arc);
+      }
+    } else {
+      for(let i = 0; i < this.majorTicks; i++) {
```

Comment:
Optionally, if you prefer declarative `map` over imperative for looping you can use [Array.prorotype.fill](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill) and [Array.prorotype.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) to get a more compact form:

```js
  public getArcs(): any {
    return Array(this.majorTicks)
      .fill(null)
      .map((_, i) => ({
        startAngle: this.getTickAngle(i),
        endAngle: this.getTickAngle(i + 1),
        innerRadius: this.dimensions.innerArcRadius,
        outerRadius: this.dimensions.outerArcRadius,
        color: this.colors.getColor(this.ticks[i])
      }));
  }
```